src/refiners/training_utils/trainer.py

```diff
@@ -356,25 +379,39 @@ def backward(self) -> None:
             self._call_callbacks(event_name="on_lr_scheduler_step_begin")
             self.lr_scheduler.step()
             self._call_callbacks(event_name="on_lr_scheduler_step_end")
+        backward_time = time.time()-start
         if self.clock.is_evaluation_step:
             self.evaluate()
+        return backward_time
 
-    def step(self, batch: Batch) -> None:
+    def step(self, batch: Batch) -> tuple[float, float]:
         """Perform a single training step."""
+        start = time.time()
         self._call_callbacks(event_name="on_compute_loss_begin")
         loss = self.compute_loss(batch=batch)
         self.loss = loss
+        forward_time = time.time() - start
+        self.forward_time_m.update(forward_time)
+        start = time.time()
         self._call_callbacks(event_name="on_compute_loss_end")
-        self.backward()
+        backward_time = self.backward(start)
+        self.backprop_time_m.update(backward_time)
+        return forward_time, backward_time
```

Comment:
Instead of making the signature of `step` more complex, maybe we could use a decorator and attributes to leave it as is